core/src/main/java/org/ehcache/core/EhcacheWithLoaderWriter.java

```diff
@@ -1019,79 +831,9 @@ public boolean replace(final K key, final V oldValue, final V newValue) throws C
     }
   }
 
-  /**
-   * {@inheritDoc}
-   */
-  @Override
-  public CacheRuntimeConfiguration<K, V> getRuntimeConfiguration() {
-    return runtimeConfiguration;
-  }
-
-  /**
-   * {@inheritDoc}
-   */
-  @Override
-  public void init() {
-    statusTransitioner.init().succeeded();
-  }
-
-  /**
-   * {@inheritDoc}
-   */
-  @Override
-  public void close() {
-    statusTransitioner.close().succeeded();
-  }
-
-  /**
-   * {@inheritDoc}
-   */
-  @Override
-  public Status getStatus() {
-    return statusTransitioner.currentStatus();
-  }
-
-  /**
-   * {@inheritDoc}
-   */
-  @Override
-  public void addHook(LifeCycled hook) {
-    statusTransitioner.addHook(hook);
-  }
-
-  void removeHook(LifeCycled hook) {
-    statusTransitioner.removeHook(hook);
-  }
-
-  private static void checkNonNull(Object thing) {
-    if(thing == null) {
-      throw new NullPointerException();
-    }
-  }
-
-  private static void checkNonNull(Object... things) {
-    for (Object thing : things) {
-      checkNonNull(thing);
-    }
-  }
-
-  private void checkNonNullContent(Collection<?> collectionOfThings) {
-    checkNonNull(collectionOfThings);
-    for (Object thing : collectionOfThings) {
-      checkNonNull(thing);
-    }
-  }
-
-  private void addBulkMethodEntriesCount(BulkOps op, long count) {
-    bulkMethodEntries.get(op).add(count);
-  }
-
-  /**
-   * {@inheritDoc}
-   */
   @Override
-  public Jsr107Cache<K, V> getJsr107Cache() {
-    return jsr107Cache;
+  public Jsr107Cache<K, V> createJsr107Cache() {
+    return new Jsr107CacheImpl();
```

Comment:
Why this change? There now can be multiple instances of that type for a given Cache. And I am not sure that does not break some (untested) spec wording. So unless you have a strong argument for this, I would tend to leave it as it was - that is a field initialised during construction.

Reply:
I can't. Because I can't this do `super(new Jsr107CacheImpl(this))`. The alternative is to set `this` using a setter. Which is ugly. But no, it doesn't break anything because the JSR wrapper will call `createJsr107Cache` once. It's even better because it is only initialize when JSR107 is used.

Reply:
Or have a field in subclasses for this ...

Reply:
Yes. That would work too. But I preferred this way. Isn't it more efficient to instantiate it when needed?